sklearn/datasets/_arff_parser.py

```diff
@@ -371,11 +371,27 @@ def _pandas_arff_parser(
         names=[name for name in openml_columns_info],
         dtype=dtypes,
     )
-
     columns_to_select = feature_names_to_select + target_names_to_select
     columns_to_keep = [col for col in frame.columns if col in columns_to_select]
     frame = frame[columns_to_keep]
 
+    # strip quotes to be consistent with LIAC ARFF
+    re_start_end_quotes = "^[\"'](.*[\"'])$"
```

Comment:
This might not be correct because it match strings such as `"something'` which are not balanced quoted values.

Also I am not sure if we should process the values in isolation or based on a column-wide metadata that should be explicit about the quotation style of the column.

Reply:
But maybe here we just aim at reproducing the behavior of the old "liac-arff" parser without aiming for correctness and clean data will be fetched by using unambiguous file format such as parquet in the future.

Reply:
> Also I am not sure if we should process the values in isolation or based on a column-wide metadata that should be explicit about the quotation style of the column.

In the last push, we can change `.any()` by `.all()` to get this behaviour for the full column.

Reply:
> This might not be correct because it match strings such as "something' which are not balanced quoted values.

So we should do a loop on the following:

```python
re_start_end_balanced_quotes = ["^[\"](.*[\"])$", "^['](.*['])$"]
```

Reply:
actually replace should take a list so it should be fine.

Reply:
For information here is the list of attributes types in ARFF files:

https://waikato.github.io/weka-wiki/formats_and_processing/arff_stable/

I think we should support:

- numeric should be left to pandas auto detection between Int64 and float64
- integer should be mapped to pandas' `"Int64"` a priori (as an argument to `pd.read_csv`)
- real should be mapped to `float64` a priori
- [nominal-specification] should be mapped to a categorical variable built appriori from the attribute declation
- string should be mapped to `str` objects
- date [date-format]: would requires proper parsing of the date-format and explicit handling in read_csv but I don't know if it's actually used in any openml dataset.

Reply:
> But maybe here we just aim at reproducing the behavior of the old "liac-arff" parser without aiming for correctness and clean data will be fetched by using unambiguous file format such as parquet in the future.

Thinking a but more about this, maybe we could do a study on the available parquet datasets on openml.org to use them as a source of ground truth to make future-proof design decisions on the behavior of the pandas parser: if we don't know if we should automatically strip or not the quotes of nominal variables, maybe we should try to load a few datasets with the pandas / pyarrow integration an look at the values in the resulting pandas dataframe.

Reply:
I think that it makes sense to be as consistent as possible with future parquet files instead of trying to stick to LIAC-ARFF indeed.